GivenThatWeWantToBuildAWindowsDesktopProject.It_builds_successfully_when_targeting_net_framework fails due to SDK version

The [test](https://github.com/dotnet/sdk/blob/main/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs#L153) fails with the output:
```
> C:\h\w\B8E70A73\p\d\dotnet.exe new wpf --no-restore
The template "WPF Application" was created successfully.
> C:\h\w\B8E70A73\p\d\dotnet.exe msbuild /t:Build C:\h\w\B8E70A73\t\dotnetSdkTests\2whhxznt.0r3\It_builds_suc---1E1E42EA\It_builds_suc---1E1E42EA.csproj /restore
Microsoft (R) Build Engine version 17.0.0-preview-21330-06+cdc5faeda for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\h\w\B8E70A73\t\dotnetSdkTests\2whhxznt.0r3\It_builds_suc---1E1E42EA\It_builds_suc---1E1E42EA.csproj (in 291 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
CSC : error CS8630: Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater. [C:\h\w\B8E70A73\t\dotnetSdkTests\2whhxznt.0r3\It_builds_suc---1E1E42EA\It_builds_suc---1E1E42EA_uuh0qbsu_wpftmp.csproj]
Exit Code: 1

```

This seems to be related to https://github.com/dotnet/templating/issues/3297 but occuring for the wpf template. Looks like https://github.com/dotnet/wpf/blob/main/packaging/Microsoft.Dotnet.Wpf.ProjectTemplates/content/WpfApplication-CSharp/Company.WpfApplication1.csproj#L9 needs to be updated with the same fix for https://github.com/dotnet/templating/pull/3318/files since net472 also defaults to C# 7.3: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

--- a/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs
+++ b/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs
@@ -149,7 +149,7 @@ namespace Microsoft.NET.Build.Tests
             getValuesCommand.GetValues().ShouldBeEquivalentTo(new[] { "true" });
         }
 
-        [WindowsOnlyRequiresMSBuildVersionFact("17.0.0.32901", Skip = "https://github.com/dotnet/sdk/issues/18800")]
+        [WindowsOnlyRequiresMSBuildVersionFact("17.0.0.32901")]
         public void It_builds_successfully_when_targeting_net_framework()
         {
             var testDirectory = _testAssetsManager.CreateTestDirectory().Path;

